core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -267,7 +268,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
 
     // Disable the background thread during tests.
-    if (!conf.contains("spark.testing")) {
+    if (!conf.contains(IS_TESTING)) {
```

Comment:
Can probably be `Utils.isTesting`.

Reply:
I don't think so, here we are checking the `SparkConf`, while in `Utils.isTesting` we check the system properties. So I don't think it is doable.

Reply:
Well, that's kinda the point. `spark.testing` is set as a system property by the build scripts. `SparkConf` just inherits system properties, which is why you can also check it there.

Reply:
(I saw your comment about the SHS tests removing the testing conf, let me take a look...)